web/src/PathBuilder.tsx

```diff
@@ -21,14 +21,14 @@ class PathBuilder {
     return `ws://${this.host}/ws/view`
   }
 
-  isLocal() {
-    return this.host.indexOf("localhost") === 0
-  }
-
   isSnapshot(): boolean {
     return this.snapId !== ""
   }
 
+  private isLocal() {
+    return this.host.indexOf("localhost") === 0
+  }
+
   private snapshotDataUrl(): string {
     let scheme = "https"
     if (this.isLocal()) {
```

Comment:
for example, why does this need include the scheme and host at all? it should be able to just use a relative path like "/api/..."